subsys/bluetooth/shell/gatt.c

```diff
@@ -85,19 +85,22 @@ static void reset_write_stats(void)
 	memset(&write_stats, 0, sizeof(write_stats));
 }
 
+/* This variable is write-locked when `(exchange_params.func != NULL)`.
+ * Must be zero-initialized when unlocked.
+ */
+static struct bt_gatt_exchange_params exchange_params;
+
 static void exchange_func(struct bt_conn *conn, uint8_t err,
 			  struct bt_gatt_exchange_params *params)
 {
 	shell_print(ctx_shell, "Exchange %s", err == 0U ? "successful" :
 		    "failed");
 
-	(void)memset(params, 0, sizeof(*params));
+	/* Release global `exchange_params`. */
+	__ASSERT_NO_MSG(params == &exchange_params);
+	exchange_params = (struct bt_gatt_exchange_params){};
```

Comment:
while neat, it is rather uncommon way of zeroing struct (at least not during variable declaration)

Reply:
I agree. But I think assignments are less error prone than using memset and memcpy because it gets rid of the manual size parameter. It's easy to botch it up the size so ends up as sizeof(void*). I'm sure it has happened to everyone.

The weird syntax is a compound literal. Compound literals are just anonymous variables. The same can be done with a named variable. But I think the compound literal is very convenient.

```c
const struct zeroed_bt_gatt_exchange_params = {};
exchange_params = zeroed_bt_gatt_exchange_params;
```

Reply:
This change is now basically just replacing `(void)memset(params, 0, sizeof(*params));` with `exchange_params = (struct bt_gatt_exchange_params){};`, and they do the same thing. I suggest to remove this change, as it doesn't change anything besides modifying more lines than what is needed.